### PR TITLE
gh action to check that ADMINS has not changed

### PR DESCRIPTION
**Pull request by asalzmann**
## Context and purpose of the change

We need to preserve ADMINS, but we change it frequently during testing. This check ensures we don't accidentally merge changes to ADMINS.

NOTE: we'll need a second PR to verify whether the ADMINS github action check works; for this PR it should definitely fail (because the file has been newly created!)

## Brief Changelog
- create admins.go, move ADMINS there
- add github action to verify admins.go has not changed

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: utils/admins.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package utils
+
+var ADMINS = map[string]bool{
```
Comment (sampocs) [inline utils/admins.go:3]: Might be a good opportunity to rename this from `ADMINS` to `Admins` to abide by the golang style guide

File: .github/workflows/verify_admins.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest  # windows-latest | macos-latest
```
Comment (sampocs) [inline .github/workflows/verify_admins.yml:13]:
```suggestion
    runs-on: ubuntu-latest  
```

File: .github/workflows/verify_admins.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    name: Verify admins.go did not change
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # OR "2" -> To retrieve the preceding commit.
```
Comment (sampocs) [inline .github/workflows/verify_admins.yml:18]:
```suggestion
          fetch-depth: 0 
```

File: .github/workflows/verify_admins.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Check if admins.go changed
+        id: admin-go-changed
+        uses: tj-actions/changed-files@v32
+        with:
+          files: |
+            utils/admins.go
+
+      - name: Run step admins.go changed
```
Comment (sampocs) [inline .github/workflows/verify_admins.yml:27]:
```suggestion
      - name: Fail if admins.go changed
```

File: .github/workflows/verify_admins.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        if: steps.admin-go-changed.outputs.any_changed == 'true'
+        run: exit 1
```
Comment (sampocs) [inline .github/workflows/verify_admins.yml:29]: First time I looked at the action output, I couldn't really tell if check failed or the action itself failed. Might help to change this to:
```suggestion
        run: echo "Admin file was changed" && exit 1
```

**Review by sampocs — APPROVED** [review]
Thanks for adding this! This will save a lot of headaches!

File: utils/admins.go (hunk 1 above)
Comment (asalzmann) [reply utils/admins.go]: done

**Pull request merged**